File: apps/mobile/lib/features/map_discovery/data/dtos/slatoki_tent_dto.dart
import "../../domain/entities/slatoki_tent.dart";

/// JSON mapping for the `SlatokiTent` schema in docs/api/openapi.yaml.
class SlatokiTentDto {
  const SlatokiTentDto({
    required this.deploymentStatus,
    required this.matCapacity,
    required this.hasLighting,
    required this.hasPrivacyCurtain,
  });

  factory SlatokiTentDto.fromJson(Map<String, dynamic> json) {
    return SlatokiTentDto(
      deploymentStatus: json["deploymentStatus"] as String,
      matCapacity: json["matCapacity"] as int,
      hasLighting: json["hasLighting"] as bool,
      hasPrivacyCurtain: json["hasPrivacyCurtain"] as bool,
    );
  }

  final String deploymentStatus;
  final int matCapacity;
  final bool hasLighting;
  final bool hasPrivacyCurtain;

  SlatokiTent toEntity() {
    return SlatokiTent(
      deploymentStatus: switch (deploymentStatus) {
        "deployed" => DeploymentStatus.deployed,
        "folded" => DeploymentStatus.folded,
        _ => DeploymentStatus.folded,
      },
      matCapacity: matCapacity,
      hasLighting: hasLighting,
      hasPrivacyCurtain: hasPrivacyCurtain,
    );
  }
}

File: apps/backend/src/modules/station-network/interface/dto/slatoki-tent.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { SlatokiTent } from '../../domain/entities/slatoki-tent.entity';

/** Matches openapi.yaml components.schemas.SlatokiTent exactly. */
export class SlatokiTentDto {
  @ApiProperty({ enum: ['deployed', 'folded'] })
  deploymentStatus!: string;

  @ApiProperty()
  matCapacity!: number;

  @ApiProperty()
  hasLighting!: boolean;

  @ApiProperty()
  hasPrivacyCurtain!: boolean;

  static fromDomain(tent: SlatokiTent): SlatokiTentDto {
    const dto = new SlatokiTentDto();
    dto.deploymentStatus = tent.deploymentStatus;
    dto.matCapacity = tent.matCapacity;
    dto.hasLighting = tent.hasLighting;
    dto.hasPrivacyCurtain = tent.hasPrivacyCurtain;
    return dto;
  }
}
